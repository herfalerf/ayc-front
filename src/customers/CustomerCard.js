import React from "react";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  CardActions,
  Typography,
} from "@material-ui/core";
import { Link } from "react-router-dom";

// Show information about a customer
//
// Is rendered by CustomerManager to show a "card" for each team member.
//
// CustomerManager -> CustomerCard
//

const CustomerCard = ({ id, name, email, phone, company }) => {
  console.debug("CustomerCard", `CustomerId=${id}`, `CustomerName=${name}`);

  return (
    <div>
      <Card>
        <CardHeader title={name} subheader={email} />
        <CardContent>
          <Typography variant="body2" component="p">
            Customer id: {id}
          </Typography>
          <Typography variant="body2" component="p">
            Company: {company}
          </Typography>
          <Typography variant="body2" component="p">
            Phone: {phone}
          </Typography>
        </CardContent>
        <CardActions>
          <Button
            variant="contained"
            component={Link}
            color="primary"
            to={`/admin/customers/${id}`}
          >
            Edit
          </Button>
        </CardActions>
      </Card>
    </div>
  );
};

export default CustomerCard;
