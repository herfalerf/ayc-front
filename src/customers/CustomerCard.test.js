import React from "react";
import { render } from "@testing-library/react";
import CustomerCard from "./CustomerCard";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <CustomerCard />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CustomerCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
