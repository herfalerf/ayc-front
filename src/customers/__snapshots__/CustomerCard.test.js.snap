// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        />
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Customer id: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Company: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Phone: 
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          href="/admin/customers/undefined"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Edit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
